inception-external-search-solr/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/solr/SolrSearchProvider.java

```diff
@@ -0,0 +1,330 @@
+            result.setTimestamp(dtf.format(now));
+        }
+
+        // Field with the entire text
+        result.setOriginalSource((String) document.getFirstValue(aTraits.getTextField()));
+
+    }
+
+
+    /**
+     * Search a Solrdocument by id and return an ExternalSearchResult.
+     * @param aRepository Solr repository
+     * @param aTraits request parameters
+     * @param aCollectionId the name of the collection
+     * @param aDocumentId the id of the document
+     * @return is used by the module external search core in order to get a preview of the document
+     * @throws SolrException Connection exception and request exception
+     */
+    @Override
+    public ExternalSearchResult getDocumentResult(DocumentRepository aRepository,
+                                                  SolrSearchProviderTraits aTraits,
+                                                  String aCollectionId,
+                                                  String aDocumentId)
+            throws IOException, SolrException {
+        if (!aCollectionId.equals(aTraits.getIndexName())) {
+            throw new IllegalArgumentException(
+                    "Requested collection name does not match connection collection name");
+        }
+
+        SolrQuery getQuery = new SolrQuery(DOC_ID_KEY + ":" + aDocumentId);
+
+        try (HttpSolrClient client = makeClient(aTraits)) {
+            ExternalSearchResult result = new ExternalSearchResult(aRepository, aCollectionId,
+                    aDocumentId);
+
+
+
+            // Send get query
+            try {
+                QueryResponse response = client.query(aTraits.getIndexName(), getQuery);
+                SolrDocumentList documents = response.getResults();
+                SolrDocument document = documents.get(0);
+                fillResultWithMetadata(result, document, aTraits);
+            } catch (SolrServerException e) {
+                e.printStackTrace();
+            }
+
+            return result;
+        }
+    }
+
+    /**
+     * Search a Solrdocument by id and return the text
+     * @param aRepository Solr repository
+     * @param aTraits Request parameters
+     * @param aCollectionId Name of the collection / index
+     * @param aDocumentId unique id of the document
+     * @return the text of the document
+     * @throws SolrException wrong id or connection issues
+     */
+    @Override
+    public String getDocumentText(DocumentRepository aRepository,
+                                  SolrSearchProviderTraits aTraits,
+                                  String aCollectionId, String aDocumentId)
+            throws IOException
+    {
+        if (!aCollectionId.equals(aTraits.getIndexName()))
+        {
+            throw new IllegalArgumentException(
+                    "Requested collection name does not match connection collection name");
+        }
+
+        SolrQuery getQuery = new SolrQuery(DOC_ID_KEY + ":" + aDocumentId);
```

Comment:
See comment above about escaping documentID and other user/system-provided values in query strings.

Reply:
Here I'm not fully sure about what we want. It's nice to have the full power of query syntax in normal case. Alexis did something to escape input in the case of querying on IDs.